ignite/services/plugin/cache_test.go

```diff
@@ -0,0 +1,110 @@
+package plugin
+
+import (
+	"net"
+	"testing"
+
+	hplugin "github.com/hashicorp/go-plugin"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPluginCacheAdd(t *testing.T) {
+	t.Run("Should cache plugin config and read from cache", func(t *testing.T) {
+		const path = "/path/to/awesome/plugin"
+		unixFD, _ := net.ResolveUnixAddr("unix", "/var/folders/5k/sv4bxrs102n_6rr7430jc7j80000gn/T/plugin193424090")
+
+		rc := hplugin.ReattachConfig{
+			Protocol:        hplugin.ProtocolNetRPC,
+			ProtocolVersion: hplugin.CoreProtocolVersion,
+			Addr:            unixFD,
+			Pid:             24464,
+		}
+
+		err := WritePluginConfigCache(path, rc)
+		require.NoError(t, err)
+
+		c := hplugin.ReattachConfig{}
+		err = ReadPluginConfigCache(path, &c)
+		require.NoError(t, err)
```

Comment:
Shouldn't you assert that rc==c ?

Reply:
it make sense to check the struct equality. has been updated.